ebos/eclproblem.hh

```diff
@@ -2091,13 +2092,19 @@ protected:
             this->mixControls_.updateLastValues(elemIdx, elemFluidState.Rs(), elemFluidState.Rv());
 
             if constexpr (enablePolymer)
-                 this->polymer_.concentration[elemIdx] = eclWriter_->eclOutputModule().getPolymerConcentration(elemIdx);
+                 this->polymer_.concentration[elemIdx] =
+                     eclWriter_->eclOutputModule().getPolymerConcentration(getLookUpData(elemIdx));
             if constexpr (enableMICP){
-                 this->micp_.microbialConcentration[elemIdx] = eclWriter_->eclOutputModule().getMicrobialConcentration(elemIdx);
-                 this->micp_.oxygenConcentration[elemIdx] = eclWriter_->eclOutputModule().getOxygenConcentration(elemIdx);
-                 this->micp_.ureaConcentration[elemIdx] = eclWriter_->eclOutputModule().getUreaConcentration(elemIdx);
-                 this->micp_.biofilmConcentration[elemIdx] = eclWriter_->eclOutputModule().getBiofilmConcentration(elemIdx);
-                 this->micp_.calciteConcentration[elemIdx] = eclWriter_->eclOutputModule().getCalciteConcentration(elemIdx);
+                 this->micp_.microbialConcentration[elemIdx] =
+                     eclWriter_->eclOutputModule().getMicrobialConcentration(getLookUpData(elemIdx));
+                 this->micp_.oxygenConcentration[elemIdx] =
+                     eclWriter_->eclOutputModule().getOxygenConcentration(getLookUpData(elemIdx));
+                 this->micp_.ureaConcentration[elemIdx] =
+                     eclWriter_->eclOutputModule().getUreaConcentration(getLookUpData(elemIdx));
+                 this->micp_.biofilmConcentration[elemIdx] =
+                     eclWriter_->eclOutputModule().getBiofilmConcentration(getLookUpData(elemIdx));
+                 this->micp_.calciteConcentration[elemIdx]
+                     = eclWriter_->eclOutputModule().getCalciteConcentration(getLookUpData(elemIdx));
```

Comment:
Changing this part here might be a bit early. This here is for restart. If you have run a simulation then you can restart at a later point again.  I don't think we have thought about this for LGRs yet. As these concentrations will be calculated on the leaf grid they probably should also be stored for the leaf grid to preserve the full details of the simulation.

Maybe we should  remove these changes and thow an exception (std::invalid_argumet) at the start of the function with a message that refined grids are not supported for restart, if the grid has more than 1 levels.

IMHO we should not change anything here.

Reply:
Thanks for the explanation. I add the exception then.